fe/fe-core/src/test/java/com/starrocks/alter/AlterLakeTableTest.java

```diff
@@ -0,0 +1,292 @@
+        Assert.assertNotNull(table.getPartition("p2"));
+        Assert.assertNotNull(table.getPartition("p3"));
+
+        dropSQL = "drop table test_lake_partition";
+        dropTableStmt = (DropTableStmt) UtFrameUtils.parseStmtWithNewParser(dropSQL, ctx);
+        GlobalStateMgr.getCurrentState().getLocalMetastore().dropTable(dropTableStmt);
+    }
+
+    @Test
+    public void testMultiRangePartitionForLakeTable() throws Exception {
+        ConnectContext ctx = starRocksAssert.getCtx();
+        String dropSQL = "drop table if exists site_access";
+        DropTableStmt dropTableStmt = (DropTableStmt) UtFrameUtils.parseStmtWithNewParser(dropSQL, ctx);
+        GlobalStateMgr.getCurrentState().getLocalMetastore().dropTable(dropTableStmt);
+        String createSQL = "CREATE TABLE site_access (\n" +
+                "    datekey INT,\n" +
+                "    site_id INT,\n" +
+                "    city_code SMALLINT,\n" +
+                "    user_name VARCHAR(32),\n" +
+                "    pv BIGINT DEFAULT '0'\n" +
+                ")\n" +
+                "DUPLICATE KEY(datekey, site_id, city_code, user_name)\n" +
+                "PARTITION BY RANGE (datekey) (\n" +
+                "    START (\"1\") END (\"5\") EVERY (1)\n" +
+                ")\n" +
+                "DISTRIBUTED BY HASH(site_id) BUCKETS 3\n" +
+                "PROPERTIES (\n" +
+                "    \"replication_num\" = \"1\"\n" +
+                ")";
+
+        CreateTableStmt createTableStmt = (CreateTableStmt) UtFrameUtils.parseStmtWithNewParser(createSQL, ctx);
+        StarRocksAssert.utCreateTableWithRetry(createTableStmt);
+        Database db = GlobalStateMgr.getCurrentState().getDb("test");
+
+        String alterSQL = "ALTER TABLE site_access \n" +
+                "   ADD PARTITIONS START (\"7\") END (\"9\") EVERY (1)";
+
+        AlterTableStmt alterTableStmt = (AlterTableStmt) UtFrameUtils.parseStmtWithNewParser(alterSQL, ctx);
+        AddPartitionClause addPartitionClause = (AddPartitionClause) alterTableStmt.getOps().get(0);
+        GlobalStateMgr.getCurrentState().getLocalMetastore().addPartitions(db, "site_access", addPartitionClause);
+
+        Table table = GlobalStateMgr.getCurrentState().getDb("test")
+                .getTable("site_access");
+
+        Assert.assertNotNull(table.getPartition("p1"));
+        Assert.assertNotNull(table.getPartition("p2"));
+        Assert.assertNotNull(table.getPartition("p3"));
+        Assert.assertNotNull(table.getPartition("p4"));
+        Assert.assertNotNull(table.getPartition("p7"));
+        Assert.assertNotNull(table.getPartition("p8"));
+
+        dropSQL = "drop table site_access";
+        dropTableStmt = (DropTableStmt) UtFrameUtils.parseStmtWithNewParser(dropSQL, ctx);
+        GlobalStateMgr.getCurrentState().getLocalMetastore().dropTable(dropTableStmt);
+    }
+
+    @Test
+    public void testAlterTableCompactionForLakeTable() throws Exception {
```

Comment:
Please move this test to `fe/fe-core/src/test/java/com/starrocks/lake/AlterTest.java`